network_plugin/__init__.py

```diff
@@ -268,7 +274,6 @@ def del_ondemand_public_ip(vca_client, gateway, ip, ctx):
         try to deallocate public ip
     """
     ctx.logger.info("Try to deallocate public IP {0}".format(ip))
-    wait_for_gateway(vca_client, gateway.get_name(), ctx)
```

Comment:
Looks as it will be regression in case several thread that tried to change Nat rules